### PR TITLE
Add is_subtype argument for metatype register 

### PR DESCRIPTION
**Pull request by AlexanderDokuchaev**
### Changes

Add `is_subtype` argument for meta type register 

### Reason for changes

Registering a nonlinear hierarchy of sub meta types.

### CODE REVIEW

File: nncf/common/graph/operator_metatypes.py [hunk 1]
```diff
@@ -100,15 +101,15 @@ def wrap(obj: Type[OperatorMetatype]) -> Type[OperatorMetatype]:
...
-            op_names = obj.get_all_aliases()
-            for name in op_names:
-                if name in self._op_name_to_op_meta_dict and not obj.subtype_check(self._op_name_to_op_meta_dict[name]):
-                    raise nncf.InternalError(
-                        "Inconsistent operator metatype registry - single patched "
-                        "op name maps to multiple metatypes!"
-                    )
-
-                self._op_name_to_op_meta_dict[name] = obj
+            if not is_subtype:
```
Comment (daniil-lyakhov) [inline nncf/common/graph/operator_metatypes.py:104]:
```suggestion
            if is_subtype:
                return obj
            ....
```

**Review by daniil-lyakhov — CHANGES_REQUESTED** [review]

File: nncf/common/graph/operator_metatypes.py (hunk 1 above)
Comment (AlexanderDokuchaev) [reply nncf/common/graph/operator_metatypes.py:104]: I'm not sure that code like this

```python
if is_subtype:
    return obj
...
return obj
```

is better that 

```python
if not is_subtype:
   ...
return obj
```

Comment (daniil-lyakhov) [reply nncf/common/graph/operator_metatypes.py:104]: It is believed that less indentation increase code readability, for example to understand how this code is working for `is_type == True` one should skip if and check nothing is called after it. But do as you wish

**Review by daniil-lyakhov — APPROVED** [review]

**Comment by AlexanderDokuchaev** [comment]
Needs to register hierarchy of subtypes like
```
        Conv2d  
     /         \ 
ModuleCond2   DepthwiseCond2d
     |    
ModuleDepthwiseCond2d
```

**Review by KodiaqQ — APPROVED** [review]

**Pull request merged**